azure-proto-core/ArmResourceOperations.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Azure;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace azure_proto_core
+{
+    public class ArmResourceOperations : ResourceOperationsBase<ArmResourceOperations, ArmResource>
+    {
+        public ArmResourceOperations(ArmClientContext context, ResourceIdentifier id) : base(context, id) { }
+
+        public override void Validate(ResourceIdentifier identifier)
+        {
+            //the id can be of any type so do nothing
+        }
+
+        public override ArmOperation<ArmResourceOperations> AddTag(string key, string value)
+        {
+            throw new NotImplementedException();
```

Comment:
We should be able to use the ResourceManagementClient to implement these

Reply:
Ill add a TODO